config/setup/startupshutdown_test.go

```diff
@@ -0,0 +1,56 @@
+package setup
+
+import (
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+
+	"github.com/mholt/caddy/config/parse"
+	"github.com/mholt/caddy/server"
+)
+
+// The Startup function's tests are symmetrical to Shutdown tests,
+// because the Startup and Shutdown functions share virtually the
+// same functionality
+func TestStartupWithBlockingCommand(t *testing.T) {
+	var startupFuncs []func() error
+
+	startupCommand := `-startup mkdir temp_dir_for_testing
+			   -startup mkdir temp_dir_for_testing &
+			   -startup highly_unlikely_to_exist_command 
+			  `
+	c := &Controller{
+		Config:    &server.Config{Startup: startupFuncs},
+		Dispenser: parse.NewDispenser("", strings.NewReader(startupCommand)),
+	}
+
+	_, err := Startup(c)
+	if err != nil {
+		t.Errorf("Expected no errors, got: %v", err)
+	}
+
+	tests := []struct {
+		shouldExecutionErr bool
+		shouldRemoveErr    bool
+	}{
+		{false, false},
+		{false, true},
+		{true, true},
+	}
+
+	for i, test := range tests {
+		err = c.Startup[i]()
+		if err != nil && !test.shouldExecutionErr {
+			t.Errorf("Test %d recieved an error of:\n%v", i, err)
+		}
+		err = os.Remove("./temp_dir_for_testing")
+		if err != nil && !test.shouldRemoveErr {
+			t.Errorf("Test %d recieved an error of:\n%v", i, err)
+		}
+
+	}
+
+	exec.Command("rm", "-r", "./temp_dir_for_testing").Start()
```

Comment:
As you can see from all these commits, I was having difficulty with Windows. I found that the middleware.SplitCommandAndArgs function in middleware/commands.go  removes the backslashes from windows paths.  I called this function with an arguement of ["mkdir", "C:\Users\appveyor\AppData\Local\Temp\1\temp_dir_for_testing_startupshutdown.go"], and I received a return value of "mkdir" and "C:UsersappveyorAppDataLocalTemp1temp_dir_for_testing_startupshutdown.go."  I am not sure if this is an issue to worry about. What do you think?